iModelCore/iModelPlatform/DgnCore/DgnMaterial.cpp

```diff
@@ -13,6 +13,33 @@ namespace dgn_ElementHandler
 
 END_BENTLEY_DGNPLATFORM_NAMESPACE
 
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+void RenderMaterial::_OnLoadedJsonProperties()
+    {
+        BeJsValue materialAssets = GetMaterialAssetsR();
+        if (materialAssets.hasMember("renderMaterial") && materialAssets["renderMaterial"].hasMember("Map"))
+            {
+            BeJsValue map = materialAssets["renderMaterial"]["Map"];
+            BeJsDocument mapToModify;
+            mapToModify.toObject();
+            map.ForEachProperty([&](Utf8CP memberName, BeJsConst memberJson)
+                {
+                if (memberJson.isNumericMember("TextureId")) 
+                    {
+                    // Fix IDs that were previously stored as 64-bit integers rather than as ID strings.
+                    int textureId = memberJson["TextureId"].GetInt();
+                    BeInt64Id id(textureId);
+                    Utf8String hexStr = id.ToHexStr();
+                    mapToModify[memberName]["TextureId"] = hexStr.c_str();

```

Comment:
Why are you extracting it as a 32-bit signed integer? The briefcase Id is in the upper 32 bits, so unless the briefcase Id is zero the Id you get back will be wrong.

You already know it's a number, so ask `GetId64` to convert that to an Id and then assign it back as a hex string.

```suggestion
                    auto textureId = memberJson["TextureId"].GetId64<DgnTextureId>;
                    mapToModify[memberName]["TextureId"] = textureId;

```